hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java

```diff
@@ -1171,6 +1181,7 @@ public final Object internalLoad(
 				event.setEntityId( null );
 				event.setInstanceToLoad( null );
 				event.setResult( null );
+				event.setShouldUnwrapProxy( unwrapProxy );
```

Comment:
I think the set should be moved before the `fireLoad( event, type );` at line 1174.